### PR TITLE
Pin boost

### PR DESCRIPTION
**Pull request by johanneskoester**
* [x] I have read the guidelines above.
* [ ] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [x] This PR does something else (explain below).

This PR does two things:

1. All (I hope) recipes depending on boost are changed such that the boost version is pinned, via a jinja2 mechanism (see GUIDELINES.md).
2. Our build system is modified, such that multiple recipes can be build in parallel jobs on travis CI. This is achieved by calculating the (i) DAG of recipes, (ii) searching for connected components (iii) distributing the connected components to separate Travis CI jobs (iv) toposorting and building each component.

Part two is very useful if many recipes are modified in a single PR (like here). Without the modification, the PR can easily exceed the build time limits on Travis.
The number of spawned jobs can be configured in the `travis.yml`.

### CODE REVIEW

File: recipes/blast/build.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
     cd $SRC_DIR/c++
     ./configure --prefix=$PREFIX
 
-    make -j4
+    make -j2
```
Comment (bgruening) [inline recipes/blast/build.sh]: We could also use `-j$(nproc)`

File: recipes/variantbam/meta.yaml [hunk 2]
```diff
@@ -4,19 +4,24 @@ package:
 
 build:
   number: 0
-  skip: True # [osx]
+  skip: True
```
Comment (bgruening) [inline recipes/variantbam/meta.yaml]: skip: True looks wrong to me

**Comment by bgruening** [comment]
Very nice improvements!

File: recipes/variantbam/meta.yaml (hunk 2 above)
Comment (johanneskoester) [inline recipes/variantbam/meta.yaml]: The variantbam build fails with an error that seems unrelated to the pinning changes. I have added the skip in order to notify the author once this is merged.

File: recipes/blast/build.sh (hunk 1 above)
Comment (johanneskoester) [inline recipes/blast/build.sh]: A plain `-j` would achieve the same I think. However, Travis provides 1.5 cores. I am not sure how this is reported via nproc.

**Comment by johanneskoester** [comment]
Because building blast takes so long, I had to temporarily increase the number of spawned jobs to 8. I will revert this to 4 once the packages have been uploaded.

**Comment by johanneskoester** [comment]
@bioconda/all please note above changes to our build system, and the new pinning mechanism.

**Comment by bgruening** [comment]
Awesome! Thanks @johanneskoester! Any reason to not leave it to 8 cores. This https://github.com/bioconda/bioconda-recipes/pull/1282 can profit from it as well.

**Comment by chapmanb** [comment]
Johannes -- awesome, thank you for the boost fix. I'll take a look at the variantbam problem and see if I can revive that recipe.

**Comment by johanneskoester** [comment]
Thanks, Brad!

**Comment by johanneskoester** [comment]
@bgruening the thing is that spawning 8 jobs per default means that we have to wait for idle jobs to complete in most cases, because the usual case is that one recipe is updated in a PR. The thing is that travis does not start all 8 jobs immediately. Hence, I thought 4 is a good compromise. We can however tune this from time to time, and first try 8 jobs for now.

**Pull request merged**